### PR TITLE
fix: gene info QA changes

### PR DESCRIPTION
**Pull request by sienacizdziel**
#### Reviewers
**Functional:** 
@seve 
**Readability:** 

---

## Changes
- fix gene info card hiding feature in controls.ts
- remove gene info card if info button is clicked while card is open
- change all instances of "Roboto Condensed" to "Inter"
- download and import "Inter" font family
- disable info button while genesets load
- move gene info analytics to separate branch for review
- if mini histogram is not visible or loading, keep gene info button on the left
- fix histogram loading height

## Notes for Reviewer
Link to QA doc [here](https://docs.google.com/document/d/1TZgUXv9QBAshJCYSjnS8Qs6bgeZ6u48awKYAkUoDzaI/edit)

### CODE REVIEW

File: client/index.html [hunk 1]
```diff
@@ -20,9 +20,9 @@
       label,
       text,
       div {
-        font-family: "Roboto Condensed", "Helvetica Neue", "Helvetica", "Arial",
-          sans-serif;
+        font-family: "Inter", "Helvetica Neue", "Helvetica", "Arial", sans-serif;
         font-size: 14px;
+        letter-spacing: -0.006em;
```
Comment (seve) [inline client/index.html:25]: why this change?

File: client/src/actions/geneset.ts [hunk 2]
```diff
@@ -56,6 +56,7 @@ export const genesetAddGenes =
     const { obsCrossfilter: prevObsCrossfilter, annoMatrix } = state;
     const { schema } = annoMatrix;
     const varIndex = schema.annotations.var.index;
+    console.log("adding gene", genes, genesetName);
```
Comment (seve) [inline client/src/actions/geneset.ts:59]: debug?

File: client/src/components/geneExpression/gene.tsx [hunk 3]
```diff
@@ -136,9 +148,12 @@ class Gene extends React.Component<Props, State> {
       isGeneInfo,
       quickGene,
       removeGene,
+      onGeneExpressionComplete,
+      isGeneExpressionComplete,
     } = this.props;
     const { geneIsExpanded } = this.state;
     const geneSymbolWidth = 60 + (geneIsExpanded ? MINI_HISTOGRAM_WIDTH : 0);
+    console.log(geneIsExpanded, gene);
```
Comment (seve) [inline client/src/components/geneExpression/gene.tsx:156]: debug

File: client/src/components/geneExpression/gene.tsx [hunk 4]
```diff
@@ -272,8 +296,14 @@ class Gene extends React.Component<Props, State> {
...
           </div>
         </div>
-        {/* @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call. */}
-        {geneIsExpanded && <HistogramBrush isUserDefined field={gene} />}
+        {geneIsExpanded && (
+          <HistogramBrush
+            // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
+            isUserDefined
+            field={gene}
+            onGeneExpressionComplete={() => {}}
```
Comment (seve) [inline client/src/components/geneExpression/gene.tsx:304]: why the noop?

File: client/src/components/geneExpression/geneSet.tsx [hunk 5]
```diff
@@ -35,16 +37,29 @@ class GeneSet extends React.Component<{}, State> {
     this.setState({ isOpen: !isOpen });
   };
 
+  onGeneExpressionComplete = (): void => {
+    const { genesetLoadCount } = this.state;
+    // @ts-expect-error ts-migrate(2339) FIXME: Property 'setGenes' does not exist on type 'Readonl... Remove this comment to see the full error message
+    const { setGenes } = this.props;
+    this.setState({ genesetLoadCount: genesetLoadCount + 1 });
+    if (genesetLoadCount + 1 === setGenes.size) {
+      this.setState({ isGeneExpressionComplete: true });
```
Comment (seve) [inline client/src/components/geneExpression/geneSet.tsx:46]: We probably don't have to use state to capture the isComplete status, just set a variable to true in `componentDidUpdate`

File: client/src/components/geneExpression/geneSet.tsx [hunk 6]
```diff
@@ -35,16 +37,29 @@ class GeneSet extends React.Component<{}, State> {
...
+    }
+  };
+
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
   renderGenes() {
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'setName' does not exist on type 'Readonl... Remove this comment to see the full error message
     const { setName, setGenes, geneIds, geneNames } = this.props;
     const setGenesNames = [...setGenes.keys()];
+    const { isGeneExpressionComplete } = this.state;
+    // this.setState({ geneCount: setGenes.keys().length });
```
Comment (seve) [inline client/src/components/geneExpression/geneSet.tsx:56]: Delete this if we don't need it anymore

File: client/src/components/geneExpression/geneSet.tsx [hunk 7]
```diff
@@ -35,16 +37,29 @@ class GeneSet extends React.Component<{}, State> {
...
     return (
       <div data-testclass="gene-set-genes">
         {setGenesNames.map((gene) => {
           const { geneDescription } = setGenes.get(gene);
           const geneId = geneIds[geneNames.indexOf(gene)];
+          console.log(gene);
```
Comment (seve) [inline client/src/components/geneExpression/geneSet.tsx:62]: debug

File: client/src/components/menubar/diffexpButtons.tsx [hunk 8]
```diff
@@ -22,6 +22,8 @@ class DiffexpButtons extends React.PureComponent {
   computeDiffExp = () => {
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'dispatch' does not exist on type 'Readon... Remove this comment to see the full error message
     const { dispatch, differential } = this.props;
+    console.log(differential.cellist1);
+    console.log(differential.celllist2);
```
Comment (seve) [inline client/src/components/menubar/diffexpButtons.tsx:26]: debug?

File: server/app/api/v3.py [hunk 9]
```diff
@@ -182,9 +182,9 @@ def add_resource(resource, url):
     add_resource(AnnotationsObsAPI, "/annotations/obs")
     add_resource(AnnotationsVarAPI, "/annotations/var")
     add_resource(DataVarAPI, "/data/var")
+    add_resource(GeneInfoAPI, "/geneinfo")
```
Comment (seve) [inline server/app/api/v3.py:185]: did changing the order do anything? If not lets revert this change

**Review by seve — CHANGES_REQUESTED** [review]

File: client/index.html (hunk 1 above)
Comment (sienacizdziel) [reply client/index.html:25]: Requested by Harley with the font change!

File: server/app/api/v3.py (hunk 9 above)
Comment (sienacizdziel) [reply server/app/api/v3.py:185]: I went ahead and changed the order after you pointed out that the routes are organized in alphabetical order, but I can move it back if you think that doesn't matter?

File: client/src/components/geneExpression/gene.tsx (hunk 4 above)
Comment (sienacizdziel) [reply client/src/components/geneExpression/gene.tsx:304]: These histograms only exist once a gene is expanded, but I wanted to just keep track of the mini histograms (calling onGeneExpressionComplete on load of each mini histogram, until they're all loaded). TS unfortunately requires I define onGeneExpressionComplete, so I put a empty function in there. Is that okay?

**Review by seve — APPROVED** [review]
